### PR TITLE
chore(spans): Remove compatibility layer after span schema change

### PR DESCRIPTION
**Pull request by phacops**
- [x] Remove `msgpack` parsing
- [x] Allow all Sentry tags set by Relay 
- [x] Remove Sentry tags mapping (as they are called in Relay)
- [x] Remove fetching `organization_id` and `retention_days` (read from the payload)
- [x] Stop validating Sentry and user tags for empty and null values (done in Relay)
- [x] Stop forcing user tags to string (done in Relay)
- [x] Add `profile_id` to Snuba span
- [x] Validate `ingest-spans` schema with `sentry-kafka-schemas` (https://github.com/getsentry/sentry-kafka-schemas/pull/180)

### CODE REVIEW

File: src/sentry/spans/consumers/process/factory.py [hunk 1]
```diff
@@ -156,50 +104,41 @@ def _process_group_raw(snuba_span: MutableMapping[str, Any], transaction: str) -
...
-    snuba_span = _process_relay_span_v0(relay_span)
+    snuba_span = _process_relay_span_v1(relay_span)
     snuba_payload = json.dumps(snuba_span).encode("utf-8")
     return KafkaPayload(key=None, value=snuba_payload, headers=[])
 
 
 def process_message(message: Message[KafkaPayload]) -> KafkaPayload:
     try:
         return _process_message(message)
+    except ValidationError as e:
```
Comment (phacops) [inline src/sentry/spans/consumers/process/factory.py:134]: @lynnagara @untitaker How should I handle that? I want to raise an `InvalidMessage` exception so we skip the message and also log the `ValidationError` to Sentry to know what's happening. Would that be the way to do it?

Comment (untitaker) [reply src/sentry/spans/consumers/process/factory.py:134]: I think the easiest in this situation would be to log manually like you do and `return message.replace(FILTERED_PAYLOAD)`, so the next `Produce` step does not actually produce it (but the offsets still get committed)

```
from arroyo.types import FILTERED_PAYLOAD
```

i think this consumer does not have a DLQ, so raising InvalidMessage will actually stop the consumer

Comment (phacops) [reply src/sentry/spans/consumers/process/factory.py:134]: What do you mean? Currently, raising `InvalidMessage` seems to do the job. Or is it what's causing the pauses we see often?

Is this `FILTERED_PAYLOAD` just a way for the `Produce` to understand it's an empty payload and we don't want to produce it but we want to commit?

**Review by jjbayer — APPROVED** [review]
CI looks very unhappy, but once the tests pass this LGTM!

File: src/sentry/spans/consumers/process/factory.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/spans/consumers/process/factory.py]: > What do you mean? Currently, raising InvalidMessage seems to do the job. Or is it what's causing the pauses we see often?

oh, i think i misremember how it works then. nevermind

InvalidMessage is still not the fastest thing in the world, it is indeed faster to directly transform to FilteredPayload. and yeah I think you have to change the mypy type

**Pull request merged**